polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -496,7 +496,11 @@ ValueCategory MLIRScanner::VisitCallExpr(clang::CallExpr *Expr) {
               LC = CE->getOperand(0);
             std::string Val = cast<llvm::GlobalVariable>(LC)->getName().str();
             return CommonArrayToPointer(ValueCategory(
-                Glob.getOrCreateGlobalLLVMString(Loc, Builder, Val),
+                Glob.getOrCreateGlobalLLVMString(
+                    Loc, Builder, Val,
+                    isa<mlir::gpu::GPUModuleOp>(Function->getParentOp())
+                        ? FunctionContext::SYCLDevice
+                        : FunctionContext::Host),
```

Comment:
Having this as a function in `utils.h` would come handy. There's another instance in `CGExpr` which you should replace by a call.

Reply:
> Having this as a function in `utils.h` would come handy. There's another instance in `CGExpr` which you should replace by a call.

Do you mean I should move `getOrCreateGlobalLLVMString` in utils.h or just this part in a function 
```
isa<mlir::gpu::GPUModuleOp>(Function->getParentOp())
                        ? FunctionContext::SYCLDevice
                        : FunctionContext::Host)
```

Reply:
Just that part. That's the common factor we can have somewhere else.

Reply:
Done